witness/README.md

```diff
@@ -1,15 +1,45 @@
-Witnessing
-============
+# Log Witnessing
 
-Witnessing is an important part of any transparent ecosystem, as it ensures that
-misbehavior on the part of logs can be prevented or detected. This is done by
-ensuring that the _checkpoints_ published by logs are _consistent_, which means 
-the log is presenting the same version of its contents to everyone and
-thus isn't carrying out a _split-view attack_.
+## Overview
 
-In terms of checkpoints, we consider witness implementations that use the
+Witnessing is an important part of log ecosystems that prevents logs from
+successfully accomplishing a _split-view attack_. A split-view attack is
+one in which the log prevents different populations with different views.
+Each of these views is append-only, and thus all clients checking consistency
+proofs are satisfied, yet the log is not fulfilling its claim of providing a
+_globally consistent_ list.
+
+For each witnessed log, a Witness will cosign log _checkpoints_ only if it
+can verify that it is consistent with the last cosigned checkpoint, i.e. that
+this new checkpoint represents only new values being appended to the log.
+Clients that acquire checkpoints signed by multiple independent witnesses can
+have more confidence that they are seeing a truly globally consistent log.
+
+## Roles
+
+We consider witnessing to be composed of three primary _roles_:
+  1. Feeder: acquires checkpoints and consistency proofs, and provides these
+     to the witness
+  2. Witness: maintains and cosigns a checkpoint for each log, and updates
+     this state to a new checkpoint only when convinced that the delta is only
+     appending new leaves
+  3. Distributor: makes cosigned checkpoints available to clients
+
+Note that this does not mean there needs to be 3 separate actors playing these roles.
```

Comment:
Another nitty nit: three instead of 3? (https://www.grammarly.com/blog/when-to-spell-out-numbers/)